rero_ils/modules/patrons/api.py

```diff
@@ -221,12 +221,21 @@ def get_number_of_loans(self):
         results = search.source().count()
         return results
 
+    def get_number_of_open_transactions(self):
+        """Get number of open transactions for this patron."""
+        trans = PatronTransaction.get_transactions_pids_for_patron(
+            self.pid, status='open')
+        return len(list(trans))
```

Comment:
I don't know how big the list could be, but normally a query with count should be faster.

Reply:
You are right. Potentially a patron could have a loooooot of open transaction. But in the fact, it's inimaginable to have more than 20 open transactions. As the method already exists to get open transactions, and this method return a generator of pid, I just used it to not write a function doing the same but returning a count.

Maybe it's a solution to add an argument to change `get_transactions_pids_for_patrons` -->
```
def get_transactions_pids_for_patrons(patron_pid, status=None, return_type='pids'):
    ....
    if return_type == 'count':
        return query.source().count
    else:
        results = query.source('pid').scan()
        for result in results:
            yield result.pid
```

Reply:
After testing, we can't use `yield` and `return` in the same function. If Python found usage of `yield` in a function it always return a generator despite if `yield` is not reached.
I will create two separate functions